### PR TITLE
Twice url fix

### PR DESCRIPTION
**Pull request by jreidinger**
manual testing in process now.

### CODE REVIEW

File: src/lib/installation/clients/inst_extrasources.rb [hunk 1]
```diff
@@ -229,7 +223,9 @@ def GetURLsToRegister(registered)
       end
 
       urls_from_control_file = Builtins.filter(urls_from_control_file) do |one_url|
-        if Builtins.contains(registered, Ops.get_string(one_url, "baseurl", ""))
+        url = one_url["base_url"] || ""
+        normalized_url = url.end_with?("/") ? url.chop : url
```
Comment (lslezak) [inline src/lib/installation/clients/inst_extrasources.rb]: I think we need to be more careful here, the trailing `/` might be important in some cases, like in `cd:/`.

It seems that we have to parse the URL and check if the trailing `/` is there. Then remove it and rebuild the URL again...

Comment (jreidinger) [inline src/lib/installation/clients/inst_extrasources.rb]: this normalized url is used only for comparison, for other purpose and later processing original is used

Comment (lslezak) [inline src/lib/installation/clients/inst_extrasources.rb]: Yes, I overlooked that, looks OK...

**Comment by shundhammer** [comment]
This was there before this pull request, but it should be fixed before this is distributed among 80+ YaST subprojects:

In the "Bug Reports" section, "if you find a problem..." is duplicated.

**Comment by shundhammer** [comment]
Other than that, LGTM.

**Comment by aschnell** [comment]
Note: Technical the URLs only differing in a trailing slash are not identical. I once had the problem with some openSUSE repository that one of the URLs was redirected while the other one was not. On the other hand AFAIR libzypp cannot distinguish the two URLs.

**Comment by lslezak** [comment]
Libzypp does not care about the URLs at all, you can even add a repository with the same URL as many times as you wish. The only result would be increased memory consumption because of the loaded duplicates and probably slower dependency solving.

**Comment by jreidinger** [comment]
thinking about slash and different url: Even when url with and without trailing slash are technically different, I cannot image case when someone intentionally use different url in extra_urls in control file and in online repo and want to have it as two separated repositories, so I propose to ignore it.

**Comment by lslezak** [comment]
I think that for handling the `control.xml` and the openSUSE repositories it is good enough. It does not fixes all possible issues but at least fixes the original bug. If there are more issues we can fix them later.

So I'm giving my LGTM here...

**Pull request merged**